dali/operators/image/resize/tensor_resize_attr.cc

```diff
@@ -0,0 +1,262 @@
+when using scales (provided or calculated).
+
+The real output size must be integer and may differ from "ideal" output size calculated as input
+(or ROI) size multiplied by the scale factor. In that case, the output size is rounded (according
+to `size_rounding` policy) and the input ROI needs to be adjusted to maintain the scale factor.
+This parameter defines which relative point of the ROI should retain its position in the output.
+
+This point is calculated as ``center = (1 - alignment) * roi_start + alignment * roi_end``.
+Alignment 0.0 denotes alignment with the start of the ROI, 0.5 with the center of the region, and 1.0 with the end.
+Note that when ROI is not specified, roi_start=0 and roi_end=size is assumed.
+
+When using 0.5 (default), the resize operation has flip invariant properties (flipping after resizing is
+mathematically equivalent to resizing after flipping).
+
+Contains as many elements as dimensions provided for sizes/scales. If only one value is provided, it is
+applied to all dimensions.)code",
+                                   std::vector<float>{0.5}, true)
+    .AddOptionalArg(
+        "axes",
+        R"code(Indices of dimensions that `sizes`, `scales`, `max_size`, `roi_start`, `roi_end` refer to.
+
+By default, all dimensions are assumed.)code",
+        std::vector<int>{})
+    .AddOptionalArg<std::string>("size_rounding",
+                                 R"code(Determines the rounding policy when using scales.
+
+Possible values are:
+* | ``"round"`` - Rounds the resulting size to the nearest integer value, with halfway cases rounded away from zero.
+* | ``"truncate"`` - Discards the fractional part of the resulting size.
+* | ``"ceil"`` - Rounds up the resulting size to the next integer value.)code",
+                                 "round")
+    .AddParent("ResizeAttrBase");
+
+
+void TensorResizeAttr::SetFlagsAndMode(const OpSpec &spec) {
+  has_sizes_ = spec.ArgumentDefined("sizes");
+  has_scales_ = spec.ArgumentDefined("scales");
+  has_max_size_ = spec.ArgumentDefined("max_size");
+  has_mode_ = spec.ArgumentDefined("mode");
+  has_axes_ = spec.ArgumentDefined("axes");
+  has_alignment_ = spec.ArgumentDefined("alignment");
+  subpixel_scale_ = spec.GetArgument<bool>("subpixel_scale");
+
+  DALI_ENFORCE(has_scales_ + has_sizes_ == 1, "Need one of ``scales`` or ``sizes``, but not both");
+
+  bool has_roi_start = spec.ArgumentDefined("roi_start");
+  bool has_roi_end = spec.ArgumentDefined("roi_end");
+  DALI_ENFORCE(has_roi_start == has_roi_end,
+               "``roi_start`` and ``roi_end`` must be specified together");
+  has_roi_ = has_roi_start && has_roi_end;
+  roi_relative_ = spec.GetArgument<bool>("roi_relative");
+
+  if (has_mode_) {
+    mode_ = ParseResizeMode(spec.GetArgument<std::string>("mode"));
+  } else {
+    mode_ = ResizeMode::Default;
+  }
+
+  if (has_axes_)
+    axes_ = spec.GetRepeatedArgument<int>("axes");
```

Comment:
Do you validate sanity of those values, like non-repeated values, in 0..ndim -1 range? Should they be monotone?

Reply:
I guess I can add a check for in-range and unique values. They don't need to be in ascending order though